library/cwm/src/lib/cwm/dialog.rb

```diff
@@ -0,0 +1,97 @@
+require "yast"
+
+Yast.import "CWM"
+
+module CWM
+  # An OOP API and the pieces missing from {YastClass::CWM#show Yast::CWM.show}:
+  # - creating and closing a wizard dialog
+  # - Back/Abort/Next buttons
+  #
+  # @see UI::Dialog
+  # @see CWM::AbstractWidget
+  class Dialog
+    include Yast::Logger
+    include Yast::I18n
+    include Yast::UIShortcuts
+
+    # @return [String,nil] Set a title, or keep the existing title
+    abstract_method :title
```

Comment:
does it make sense to have method that return nil? like `def title; nil; end`? why not having nil as default?